docs/api/networkinterfaces/networkinterface.md

```diff
@@ -15,11 +15,11 @@ NetworkInterface is also the controlling API that the application uses to specif
 
 When you create a network interface, it starts from the disconnected state. When you call `NetworkInteface::connect()`, the interface stays connected until you call `NetworkInterface::disconnect()`. The following diagram illustrates the state changes:
 
-<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/NetworkinterfaceStates.png)<span>Network states</span></span>
+<span class="images">![](../../images/NetworkinterfaceStates.png)<span>Network states</span></span>
 
-The interface handles all state changes itself between `Connecting`, `Local connectivity` and `Global route found`. Calling `NetworkInterface::connect()` might return when either local or global connectivity states are reached. This depends on the connectivity. For example, Ethernet and Wi-Fi interfaces return when global connectivity is reached. 6LoWPAN-based mesh networks depend on the standard you're using. 6LoWPAN-ND returns when it connects to a border router that provides global connection. The thread returns when you create a local mesh network and may later get global connection when it finds a border router.
+The interface handles all state changes itself between `Connecting`, `Local connectivity` and `Global route found`. Calling `NetworkInterface::connect()` might return when either local or global connectivity states are reached. This depends on the connectivity. For example, Ethernet and Wi-Fi interfaces return when global connectivity is reached. 6LoWPAN-based mesh networks depend on the standard you're using. The `LoWPANNDInterface` returns from `connect()` call when it connects to a border router that provides global connection. The `ThreadInterface` returns from `connect()` call when it joins a local mesh network. It may later get global connection when it finds a border router.
```

Comment:
Better some questions though.

"might return" when might it not return?

"This depends on the connectivity" I can assume this is referring to WiFi, Ethernet, 6LoWPAN, Cellular, etc.
However does "connectivity" correlate to a particular network layer and would it be useful to reference said layer?

Reply:
To answer my own connectivity question. No? Because Thread sprawls multiple layers and this is TCP/IP.

Reply:
Yes.
"Connectivity" here refers to L2/L3 combination. In more specifically, it refers to which `NetworkInterface` subclass you have selected as those examples show.